resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -58,14 +58,24 @@ import org.apache.spark.util.{CallerContext, Utils}
 private[spark] class Client(
     val args: ClientArguments,
     val hadoopConf: Configuration,
-    val sparkConf: SparkConf)
+    val sparkConf: SparkConf,
+    val sysEnvironment: scala.collection.immutable.Map[String, String])
```

Comment:
So, when I requested that you made up your mind about whether to support custom envs or not, *this* is what I meant. You can revert all changes to this file that now use `sysEnvironment` instead of `sys.env`, including the new constructor that takes an env map and the new parameter to `populateClasspath`.